controllers/sensor/operator.go

```diff
@@ -100,86 +102,114 @@ func (soc *sOperationCtx) operate() error {
 		// check if sensor job is already been created for repeated sensor
 		_, err = soc.controller.kubeClientset.BatchV1().Jobs(soc.s.Namespace).Get(soc.s.Name, metav1.GetOptions{})
 		if err != nil {
-			sensorJob := &batchv1.Job{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      soc.s.Name,
-					Namespace: soc.s.Namespace,
-					Labels: map[string]string{
-						common.LabelJobName: soc.s.Name,
+			if apierr.IsNotFound(err) {
+				// default env variables
+				envVars := []corev1.EnvVar{
+					{
+						Name:  common.SensorName,
+						Value: soc.s.Name,
 					},
-					OwnerReferences: []metav1.OwnerReference{
-						*metav1.NewControllerRef(soc.s, v1alpha1.SchemaGroupVersionKind),
+					{
+						Name:  common.SensorNamespace,
+						Value: soc.s.Namespace,
 					},
-				},
-				Spec: batchv1.JobSpec{
-					Template: corev1.PodTemplateSpec{
-						ObjectMeta: metav1.ObjectMeta{
-							GenerateName: common.DefaultSensorJobName(soc.s.Name),
-							Labels: map[string]string{
-								common.LabelJobName: soc.s.Name,
-							},
+					{
+						Name:  common.SensorControllerInstanceIDEnvVar,
+						Value: soc.controller.Config.InstanceID,
+					},
+				}
+				// user defined environment variable. This feature is added due to Issue #103
+				if soc.s.Spec.EnvVars != nil {
```

Comment:
could you potentially wire in env vars from the sensor CRD in here? looking at convo in #103